### PR TITLE
Put audio system back

### PR DESCRIPTION
**Pull request by stilnat**
# Summary

credits to @EikoBiko #899 for the original PR. This PR intends to put the audio manager back into SS3D. 

- Added a bunch of sounds
- updated some items to make sound using the new system
- Added a proper audio mixer to tune SFX, ambience and music sounds independently.
- Added back the whole ambience system, pretty much untouched.
- Added back the noise collision, just changed to fit the new framework.
- Added network syncing (that's new)
- Added music support for the audio manager (that's new)

#### PR checklist
- [x] The game builds properly without errors.

- [x] No unrelated changes are present.

- [x] No "trash" files are committed.

- [x] Relevant code is documented.

- [ ] Update the related GitBook document, or create a new one if needed.

# Pictures/Videos)

Messing around with everything with sound in the PR : 

https://github.com/RE-SS3D/SS3D/assets/14344825/fabf51ac-04f9-45b5-86b6-b2381fcbaaf1

Messing around with ambient sounds : 

https://github.com/RE-SS3D/SS3D/assets/14344825/8d96df95-1d66-4467-bf2c-1c5019bd72e1

# Testing

Test the sound syncing by muting host, performing action with host and checking that client get the sound.

#### Networking checklist

- [x] Works from host in host mode.

- [x] Works from server in server mode.

- [x] Works on server in client mode.

- [x] Works and is syncronized across different clients.

- [x] Is persistent.

# Changes

- The audioManager (audioSystem now) is now supporting music as well. in the previous PR, it only supported SFX. For it to support music, I just added a new List of available audio sources, set up to use the main mixer (instead of the SFX one).
I add to restructure the code a lot to make it clean. Also now you have to precise through an enum value when you call play sound, if that should play as music or sfx. Not much of a hassle imo.

- Audio is now synced between clients, on the old code version, I'm confident that code was only local, I tried by building outdated and putting one instance of the game mute and trying to trigger sounds from it for the other client. It does not work.

>

# Known issues

- "The human prefab is no longer the parent of the listener component. Instead, the Player Camera is the parent, and it positions the listener relative to its target." Something I did not touch in this PR, I'm not sure how to deal with this, but it's probably good to do so. Need help.

- the networking part assumes there's not more than 1 audio source per network object. It uses network object to find a given network audio source. Not easy to fix, maybe better to work around it since we can play multiple clips on a single audio source https://docs.unity3d.com/ScriptReference/AudioSource.PlayOneShot.html

### CODE REVIEW

File: Assets/Content/Addressables/AssetGroups/Audios.asset [hunk 1]
```diff
@@ -0,0 +1,65 @@
...
+MonoBehaviour:
+  m_ObjectHideFlags: 0
+  m_CorrespondingSourceObject: {fileID: 0}
+  m_PrefabInstance: {fileID: 0}
+  m_PrefabAsset: {fileID: 0}
+  m_GameObject: {fileID: 0}
+  m_Enabled: 1
+  m_EditorHideFlags: 0
+  m_Script: {fileID: 11500000, guid: bbb281ee3bf0b054c82ac2347e9e782c, type: 3}
+  m_Name: Audios
```
Comment (cosmiccoincidence) [inline Assets/Content/Addressables/AssetGroups/Audios.asset]: The word "Audio" is already plural, thus the word "Audios" sounds very weird in American English. I'd recommend changing this to either "Audio" or "Sounds"

Comment (stilnat) [reply Assets/Content/Addressables/AssetGroups/Audios.asset]: go for sounds then

File: Assets/Art/Models/Furniture/Generic/jukebox.fbx
Comment (cosmiccoincidence) [inline Assets/Art/Models/Furniture/Generic/jukebox.fbx:1]: Rename jukebox.fbx and all audio assets to PascalCase.

I recommend using a script. Here is the one I was using for a while: https://github.com/Gadiguibou/stdrename

File: Assets/Content/Addressables/AssetGroups/Audios.asset (hunk 1 above)
Comment (cosmiccoincidence) [reply Assets/Content/Addressables/AssetGroups/Audios.asset]: Be sure to do this update it in all your files.

File: Assets/Art/Models/Furniture/Generic/jukebox.fbx
Comment (stilnat) [reply Assets/Art/Models/Furniture/Generic/jukebox.fbx:1]: done

File: Assets/Content/Addressables/AssetGroups/Audios.asset (hunk 1 above)
Comment (stilnat) [reply Assets/Content/Addressables/AssetGroups/Audios.asset]: done

File: Assets/Art/Sound/World/Collisions/Grillehit.ogg
Comment (cosmiccoincidence) [inline Assets/Art/Sound/World/Collisions/Grillehit.ogg:1]: GrilleHit.ogg

File: Assets/Art/Sound/World/Collisions/Trayhit1.ogg
Comment (cosmiccoincidence) [inline Assets/Art/Sound/World/Collisions/Trayhit1.ogg:1]: TrayHit1.ogg

File: Assets/Art/Sound/World/Collisions/Trayhit2.ogg
Comment (cosmiccoincidence) [inline Assets/Art/Sound/World/Collisions/Trayhit2.ogg:1]: TrayHit2.ogg

Comment (stilnat) [reply Assets/Art/Sound/World/Collisions/Trayhit2.ogg:1]: done

File: Assets/Art/Sound/World/Collisions/Trayhit1.ogg
Comment (stilnat) [reply Assets/Art/Sound/World/Collisions/Trayhit1.ogg:1]: done

File: Assets/Art/Sound/World/Collisions/Grillehit.ogg
Comment (stilnat) [reply Assets/Art/Sound/World/Collisions/Grillehit.ogg:1]: done

File: Assets/Content/Data/Audios.asset [hunk 2]
```diff
@@ -0,0 +1,28 @@
...
+MonoBehaviour:
+  m_ObjectHideFlags: 0
+  m_CorrespondingSourceObject: {fileID: 0}
+  m_PrefabInstance: {fileID: 0}
+  m_PrefabAsset: {fileID: 0}
+  m_GameObject: {fileID: 0}
+  m_Enabled: 1
+  m_EditorHideFlags: 0
+  m_Script: {fileID: 11500000, guid: 3c9b6c8ad6324c838c1743c2a2e2abc4, type: 3}
+  m_Name: Audios
```
Comment (cosmiccoincidence) [inline Assets/Content/Data/Audios.asset]: Sounds
(File name too)

Comment (stilnat) [reply Assets/Content/Data/Audios.asset]: done

**Review by cosmiccoincidence — APPROVED** [review]
Tested between host and client, seems to be working good.

**Comment by cosmiccoincidence** [comment]
Please remember to add gitbook documentation on this.

Also, Im tasking you with turning the known issues into github issues (including the lobby ambience sound one) since you know a bit more about the issues than me.

**Pull request merged**